build.gradle

```diff
@@ -147,18 +155,26 @@ subprojects {
 
     publishing {
         publications {
-            mavenJava(MavenPublication) {
+            shadow(MavenPublication) {
                 from components.java
 
-                // These are for unpublished test classes
-                suppressPomMetadataWarningsFor('testFixturesApiElements')
-                suppressPomMetadataWarningsFor('testFixturesRuntimeElements')
+                project.shadow.component(it)
 
                 // Customize artifact ID for core project
                 if (project.name == 'opensearch-remote-metadata-sdk-core') {
                     artifactId = 'opensearch-remote-metadata-sdk'
                 }
 
+                artifacts.clear()
+
+                artifact shadowJar
+                artifact sourcesJar
+                artifact javadocJar
+
+                // These are for unpublished test classes
+                suppressPomMetadataWarningsFor('testFixturesApiElements')
+                suppressPomMetadataWarningsFor('testFixturesRuntimeElements')
```

Comment:
The test fixture dependencies aren't in the pom metadata which causes warnings. But since these are only needed for tests which are not published, we can suppress the warnings.